include/tvm/tir/op.h

```diff
@@ -552,6 +552,24 @@ TVM_DLL PrimExpr trunc(PrimExpr x);
  */
 TVM_DLL PrimExpr LargeUIntImm(DataType dtype, int64_t low, int64_t high);
 
+/*!
+ * \brief Execute a multiplication between two Q-numbers x and y
+ * followed by a right shift s. The mathematical expression is:
+ *
+ *    out = round(x*y*2^-s)
+ *
+ * More about Q-numbers here: https://en.wikipedia.org/wiki/Q_(number_format)
+ *
+ * The rounding rule is to the nearest value, rounding half up
+ * (i.e., round(x.1) = x and round (x.5) = x+1)
+ * \param x first Q-number
+ * \param y second Q-number
+ * \param q Q-ness of x and y
```

Comment:
You could say that `q` is the number of fractional bits in `x` and `y`.  We should also require that `q > 0`, or otherwise the code in [intrin_rule.cc at lines 151-152](https://github.com/apache/incubator-tvm/pull/5980/files#diff-df842e89e6e7af56d4566cee5d38f7f3R151-R152) won't work.

Reply:
Agreed, number of fractional bits is better description

Reply:
Are number of fractional bits same for x and y and thats why we need only input? Lets make the description more clear.
IIUC, one can think of it using this op to perform something like Q1.31  * Q2.30. But, I think this op is restrictive than that. If it is, then lets mention it.

Reply:
I added the constraint of `q>0` in the doc string of the function (both in python and c++). The problem is that this intrinsic is vectorized and `q` becomes a vector. In order to check `q>0` within the intrinsic, I would need to go through a reduction (which seems a bit expensive for a simple check).